### PR TITLE
Add AudioReports to the Django admin

### PR DESCRIPTION
**Pull request by stacimc**
## Fixes

Related to https://github.com/WordPress/openverse-catalog/issues/354

## Description

When an image is reported by a user through the `Report Content` feature, the reports become available for review in the `Image reports` admin page (http://localhost:8000/admin/api/imagereport/). When an audio file is reported, a similar report is created in the DB, but it isn't easily accessible through the Django Admin.

This PR adds an `Audio reports` admin page. It pulls out the existing code into an abstract MediaReportAdmin that can be used as additional models are added.

## Testing Instructions

Visit both [Image](http://localhost:8000/admin/api/imagereport/) and [Audio](http://localhost:8000/admin/api/audioreport/) report admin pages and verify they work as normal (described below).

If you don't have any Image or Audio reports, you'll probably want to make some sample data. You can do this locally by going to the list view (eg `localhost:8000/v1/images`) and grabbing the id for a couple of the listed records, then POSTing a report (eg `localhost:8000/v1/<id>/report`). The process is the same for audio.

* You should see all the relevant fields on the main admin page
* You should be able to filter by status or reason
* You should be able to search by description and identifier
* When you click on the `status` for a row, you should be taken to its update page
    * If the `status` is `pending_review`, you should be able to edit the `status` field; if not, it should not be editable. No other fields should be editable in any case.
    * Test that you can save

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/catalog/api/admin.py [hunk 1]
```diff
@@ -10,20 +11,20 @@
...
-class ImageReportAdmin(admin.ModelAdmin):
-    list_display = ("reason", "status", "image_url", "description", "created_at")
+class AbstractMediaReportAdmin(admin.ModelAdmin):
     list_filter = ("status", "reason")
     list_display_links = ("status",)
     search_fields = ("description", "identifier")
     actions = None
 
+    class Meta:
+        abstract = True
```
Comment (dhruvkb) [inline api/catalog/api/admin.py:21]: I don't think `ModelAdmin` classes need `abstract = True` in their meta. Removing this seems to have no effect on the outcome.

**Review by dhruvkb — APPROVED** [review]
Looks good, works well.

File: api/catalog/api/admin.py (hunk 1 above)
Comment (stacimc) [reply api/catalog/api/admin.py:21]: Agreed. The MediaReport model itself is abstract but it's not needed in the model admin, and I don't imagine we'll have any use for it later. Removed :)

**Review by zackkrida — APPROVED** [review]
Looks great! Reported some audio with

```shell
http get 'localhost:8000/v1/audio/?format=json' | jq -r '.results[].id' | xargs -I{} http post localhost:8000/v1/audio/{}/report/ reason=mature description=zack --ignore-stdin
```

image


Then set one to be `deindexed` in the UI, which appears to have made the correct ES calls:

```
[20/May/2022 00:21:43] "GET /admin/api/audioreport/19/change/ HTTP/1.1" 200 11356
[20/May/2022 00:21:43] "GET /admin/jsi18n/ HTTP/1.1" 200 3343
[2022-05-20 00:21:48,352 - elasticsearch - 265][INFO] DELETE http://es:9200/audio/_doc/79 [status:200 request:0.010s]
[2022-05-20 00:21:48,362 - elasticsearch - 265][INFO] POST http://es:9200/audio/_refresh [status:200 request:0.010s]
```

**Review by obulat — APPROVED** [review]

File: api/catalog/api/admin.py [hunk 2]
```diff
@@ -10,20 +11,20 @@
...
-    list_display = ("reason", "status", "image_url", "description", "created_at")
+class MediaReportAdmin(admin.ModelAdmin):
     list_filter = ("status", "reason")
     list_display_links = ("status",)
     search_fields = ("description", "identifier")
     actions = None
 
+    class Meta:
+        abstract = True
+
```
[the remark this answers is not in the document]
Comment (stacimc) [reply api/catalog/api/admin.py:22]: Did I not? Oh my gosh, that was my intention. Too much switching between branches, how embarrassing 🤣 Glad you looked at it again.

**Pull request merged**